### ISSUE TITLE
Cassandra backend should use LOCAL_QUORUM for reads & writes

### ISSUE THREAD

**Issue by rustyrazorblade** [thelastpickle/cassandra-reaper#199]
There's currently very little benefit to using LOCAL_ONE for reads and writes to the cassandra backend.

* Very little performance overhead given the infrequency of calls
* In a case where there is inconsistent data in the reaper keyspace, using L_ONE could be missing clusters or schedules when the operator checks the page.  This makes reaper look broken.
* In a case where we can't satisfy L_QUORUM we won't be able to do repairs anyways

**Comment by michaelsembwever** [comment]
TL;DR, i've got a number of questions:

 - is this mission-critical?
 - what are the advantages it gives us versus the limited horizon?
 - how much work is involved in switching?
 - what are the performance implications beyond the internals to c* requests?
 - where did this stem from? and what about (and why) all the work recently done to make it eventually consistent?
 - why are we talking about all cql requests rather than specific tables?


Long version:

So i'm not for this move for a number of reasons, although I could be convinced otherwise, especially on a per prepared statement (or table) basis.
Taking note that effort has already been put in to ensure each prepared statement is using the correct consistency level and idempotence.

The codebase currently reads and writes too often to the storage within each restful api request.
By moving the code to (at least) work with LOCAL_ONE/ONE we've managing to clean up and minimise these repeated calls to the backend.
I think that this makes the "little performance overhead" discussion far more complicated. Yes the difference between QUORUM and ONE against RF=3 for a low traffic application like reaper is relatively small, but the codebase has at times thousands of unnecessary cql requests and the move to ONE ensures such things are improved. Lots of work was put into making the model async, and related to that we’re seeing *huge* performance improvements, not because of the c* performance difference, but because the service layer is being forced to stop relying upon the storage backend thousands of times per request.

It is true that on an unhealthy cluster with downed nodes that certain requests could force the user to refresh their webpage to see a newly added schedule. Take note that this won't happen for newly added clusters or repairs because they do use LOCAL_QUORUM already. Regardless of the chosen consistency level for schedules, I'm not comfortable extrapolating the decision onto all prepared statements and tables.

There are use-cases where the notion that 50% unavailability in the c* cluster means an unusable reaper interface doesn't hold true.
One of the ideas around chosen consistency levels was that it better supported the idea of co-located reaper installations (that is where a reaper instance is bundled next to every c* node). In this example we could well develop a “everywhere” RF coupled with a localhost connection. In combination with alex’s work on repairing specific hosts, or running compactions on specific nodes, etc, this would mean reaper had full availability and was only ever reliant upon its c* peer node. If we want to keep this on the horizon then the code needs to be more LOCAL_ONE friendly than it currently is, as described above. And accepting a few minor issues (like the user maybe having to refresh a page in the rare occurrence on an unhealthy cluster) might be considered better if it gives that flexibility in what reaper can do as an admin tool. This is just _one example_ on how the choice of consistency impacts us. It's not a blocker in any means, but i hope it shows an engaged discussion is warranted and it certainly isn' obvious.

The work involved in switching to LOCAL_QUORUM across the board is involved, as we'd need to switch sync model and should adjust (tighten) tests accordingly.

More importantly, is this ticket mission-critical to the roadmap towards "1.0"? 

And why weren't we having this discussion when we recently did this work in https://github.com/thelastpickle/cassandra-reaper/pull/129 (also part of the ft branch/pull-request)?

I think it's also worth while pointing out that the bug that this conversation arose from (afaik), around a failing test because of its use of eventual consistency, was quickly easily fixed with just ~15 lines of code.

**Comment by adejanovski** [comment]
Let's close this issue as integration tests are properly passing lately and we had no feedback suggesting consistency issues.

We'll open a new issue if the problem pops up.
